db/migrate/20200615134519_add_title_and_description_to_saved_scenarios.rb

```diff
@@ -0,0 +1,45 @@
+class AddTitleAndDescriptionToSavedScenarios < ActiveRecord::Migration[5.2]
+  def up
+    add_column :saved_scenarios, :title, :string
+    add_column :saved_scenarios, :description, :text
+
+    # Get current titles and descriptions from ETEngine
+    saved_scenarios = SavedScenario.all
+    total = saved_scenarios.size
+    done = 0
+    failed = 0
+    puts "Updating #{total} saved scenarios by extracting details from ETEngine"
+
+    # ETEngine only allows batches of maximum length 2048
+    saved_scenarios.in_groups_of(2048, false) do |chunked_saved_scenarios|
```

Comment:
Good to know! I started with sending them all at once, and got an error from etengine telling me I couldn't send more than 2048.

I did a small benchmark, and batches of 100 are indeed over 4 times faster than batches of 1000. Funny enough, if I don't do any batch loading the update takes only 2.5 times longer than for batches of 100. Guess that has to do with memory?

Reply:
> and got an error from etengine telling me I couldn't send more than 2048.

I was curious about this, because I couldn't remember ETEngine imposing a limit on the number of scenarios that could be batch loaded. Apparently it's Puma which refuses a request whose path is greater than 2048 characters:

```
2020-06-18 13:15:15 +0100: HTTP parse error, malformed request (): #<Puma::HttpParserError: HTTP element REQUEST_PATH is longer than the 2048 allowed length (was 13306)>
```

That log entry came while running the migration with a batch size of 2048. The batch load still fails, and the result is each call to `SavedScenario#scenario` fetches the individual scenario. 100 scenarios at once should keep us well below the 2048 character limit. Fixing this results in a faster migration as well (10s on my machine vs 37s).

Reply:
Ah that makes sense! Strange I didn't pick that up... Thanks for investigating for me!